src/scripts/main.js

```diff
@@ -1 +1,77 @@
 'use strict';
+
+(() => {
+  const body = document.querySelector('body');
+
+  document.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+  });
+
+  const notification = (parentNode, classes, text) => {
+    const node = document.createElement('div');
+
+    node.classList.add(...classes);
+    node.textContent = text;
+    node.dataset.qa = 'notification';
+    parentNode.append(node);
+  };
+
+  const firstPromise = new Promise((resolve, reject) => {
+    document.addEventListener('click', () => {
+      resolve('First promise was resolved');
+    });
+
+    setTimeout(() => {
+      reject(new Error('First promise was rejected'));
+    }, 3000);
+  });
+
+  const secondPromise = new Promise((resolve) => {
+    const handleClick = (e) => {
+      e.preventDefault();
+
+      resolve('Second promise was resolved');
+    };
```

Comment:
It's a code quality issue. You are using the same function for both click and contextmenu events. This might not be the best approach as it can lead to unexpected behavior. It would be better to have separate handlers for different events. This will make your code more readable and maintainable.